service/src/main/java/uk/gov/hmcts/reform/fnp/model/fee/FeeType.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.fnp.model.fee;
+
+import com.google.common.collect.ImmutableList;
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+
+public enum FeeType {
+    // Names should match OrderType enum names exactly
+    C2_WITHOUT_NOTICE,
+    C2_WITH_NOTICE,
+    CARE_ORDER,
+    EDUCATION_SUPERVISION_ORDER,
+    EMERGENCY_PROTECTION_ORDER,
+    INTERIM_CARE_ORDER,
+    INTERIM_SUPERVISION_ORDER,
+    OTHER,
+    PLACEMENT,
+    SUPERVISION_ORDER;
+
+    private static final Map<OrderType, FeeType> orderToFeeMap = buildOrderFeeMap();
+
+    public static FeeType fromOrderType(OrderType orderType) {
+        return orderType != null ? orderToFeeMap.get(orderType) : null;
```

Comment:
Shouldn't be, currently it is only called from the overloaded method, and that list that is passed to it shouldn't have any nulls in it. Do you think I should remove the check? Could add `@Nonnull` to make the requirement more specific.

Reply:
If we cannot come up with valid scenario when `orderType` is null, I'd suggest to remove the check. We want our app to blow up with NPE, because it means that something is seriously wrong.

Reply:
Also, just noticed that this method is not used anywhere (apart from unit test), so we should probably remove it.

Reply:
Good point, removed method and altered tests